app/controllers/users_controller.rb

```diff
@@ -0,0 +1,51 @@
+class UsersController < ApplicationController
+  get '/signup' do
+    if !logged_in?
+      erb :'users/create_user'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  post '/signup' do
+    user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    if !user.username.empty? && !user.email.empty? && user.save
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.